src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -0,0 +1,198 @@
+void fill_tensor(ov::SoPtr<ov::ITensor> tensor, T fill_val, size_t offset = 0u) {
+    T* tensor_data = tensor->data<T>();
+    std::fill(tensor_data + offset, tensor_data + tensor->get_size(), fill_val);
+}
+
+ov::SoPtr<ov::ITensor> make_tensor_slice(ov::SoPtr<ov::ITensor> tensor,
+                                         uint32_t dim,
+                                         uint32_t start_pos,
+                                         uint32_t end_pos) {
+    ov::Shape start_shape(std::vector<size_t>(tensor->get_shape().size(), 0u));
+    start_shape[dim] = start_pos;
+    ov::Shape end_shape = tensor->get_shape();
+    end_shape[dim] = end_pos;
+    return ov::get_tensor_impl(ov::Tensor(ov::make_tensor(tensor), start_shape, end_shape));
+}
+
+ov::npuw::LLMInferRequest::LLMInferRequest(const std::shared_ptr<ov::npuw::LLMCompiledModel>& compiled_model,
+                                           const ov::npuw::LLMCompiledModel::KVCacheDesc& kvcache_desc)
+    : ov::ISyncInferRequest(compiled_model),
+      m_kvcache_desc(kvcache_desc) {
+    LOG_VERB(__PRETTY_FUNCTION__);
+    LOG_BLOCK();
+    m_kvcache_request = compiled_model->m_kvcache_compiled->create_infer_request();
+    m_prefill_request = compiled_model->m_prefill_compiled->create_infer_request();
+
+    for (auto input_port : m_prefill_request->get_compiled_model()->inputs()) {
+        m_prefill_in_ports.emplace(input_port.get_any_name(), input_port);
+    }
+    for (auto output_port : m_prefill_request->get_compiled_model()->outputs()) {
+        m_prefill_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
+
+    for (auto input_port : m_kvcache_request->get_compiled_model()->inputs()) {
+        m_kvcache_in_ports.emplace(input_port.get_any_name(), input_port);
+    }
+    for (auto output_port : m_kvcache_request->get_compiled_model()->outputs()) {
+        m_kvcache_out_ports.emplace(output_port.get_any_name(), output_port);
+    }
+    LOG_VERB("Done");
+}
+
+void ov::npuw::LLMInferRequest::prepare_for_new_conversation() {
+    // FIXME: for input_ids it must be padding from tokenizer that not available from here
+    // Get it from NPUW options
+    LOG_VERB(__PRETTY_FUNCTION__);
+    LOG_BLOCK();
+
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("input_ids")), 0u);
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("attention_mask")), 0u);
+    fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at("position_ids")), 0u);
+    fill_tensor<int64_t>(m_kvcache_request->get_tensor(m_kvcache_in_ports.at("attention_mask")), 0u);
+    m_kvcache_desc.num_stored_tokens = 0u;
+
+    LOG_VERB("Done");
+}
+
+void ov::npuw::LLMInferRequest::infer_prefill(ov::SoPtr<ov::ITensor> input_ids,
+                                              ov::SoPtr<ov::ITensor> attention_mask,
+                                              ov::SoPtr<ov::ITensor> position_ids) {
+    LOG_VERB(__PRETTY_FUNCTION__);
```

Comment:
Please don't do this everywhere

Reply:
After all, every LOG_ method can put its __PRETTY_FUNCTION__ as part of the log header. Write what you're going to do, not the method name - that's the approach

Reply:
Fixed